languages/js/src/helpers.ts

```diff
@@ -217,6 +225,36 @@ function parseExternalIsa({
   };
 }
 
+/**
+ * Try to parse a JSON payload received from across the WebAssembly boundary as
+ * an [[`ExternalOp`]].
+ *
+ * @internal
+ */
+function parseExternalOp({
+  call_id: acallid,
+  args,
+  operator,
+}: obj): QueryEvent {
+  if (!Number.isSafeInteger(acallid)) {
+    throw new Error();
+  }
+
+  args.forEach((arg: unknown, narg: number) => {
+    if (!isPolarTerm(arg)) {
+      throw new Error();
+    }
+  });
+  return {
+    kind: QueryEventKind.ExternalOp,
+    data: {
+      args,
+      callId: acallid,
+      operator,
```

Comment:
I think this function would be a good place to validate that `operator` is a variant of the `PolarOperator` enum since we're doing all of our other "did we receive bad data" checks here.

If we make that change, then we can remove the `default:` fallback in the switch statement in `src/Host.ts` and throw that `PolarError` here instead.